www/js/metrics.js

```diff
@@ -421,14 +421,27 @@ angular.module('emission.main.metrics',['nvd3', 'emission.services', 'ionic-date
           // If there is no data from last week (ex. new user) 
           // Don't store the any other data as last we data
          }
+         var seventhDayAgo = moment().utc().startOf('day').subtract(7, 'days');
+         var twoWeeksAgoDuration = [];
+         var twoWeeksAgoMedianSpeed = [];
+         var twoWeeksAgoDistance = [];
+         var userDuration = [];
+         var usedMedianSpeed = [];
+         var userCount = [];
+         var userDistance = [];
         if(first){
-          var twoWeeksAgoDuration = results[0].user_metrics.slice(0,7); 
-          var twoWeeksAgoMedianSpeed = results[1].user_metrics.slice(0,7);
-          var twoWeeksAgoDistance = results[3].user_metrics.slice(0,7)
-          var userDuration = results[0].user_metrics.slice(7);
-          var usedMedianSpeed = results[1].user_metrics.slice(7);
-          var userCount = results[2].user_metrics.slice(7);
-          var userDistance = results[3].user_metrics.slice(7);
+          for(var i in results[0].user_metrics) {
+            if(seventhDayAgo.isSameOrBefore(moment.unix(results[0].user_metrics[i].ts).utc())){
+              userDuration.push(results[0].user_metrics[i]);
+              usedMedianSpeed.push(results[1].user_metrics[i]);
+              userCount.push(results[2].user_metrics[i]);
+              userDistance.push(results[3].user_metrics[i]);  
+            } else {
+              twoWeeksAgoDuration.push(results[0].user_metrics[i]);
+              twoWeeksAgoMedianSpeed.push(results[1].user_metrics[i]);
+              twoWeeksAgoDistance.push(results[3].user_metrics[i]);
+            }
+          }
           var aggDuration = results[0].aggregate_metrics.slice(7);
           var aggMedianSpeed = results[1].aggregate_metrics.slice(7);
           var aggCount = results[2].aggregate_metrics.slice(7);
```

Comment:
why not do something similar for the aggregate metrics as well? I agree this is less important because the aggregate metrics are unlikely to be missing, but it doesn't cost anything to handle that corner case as well.